README.md

```diff
@@ -1,10 +1,47 @@
+# p4c: A Comprehensive P4 Compiler
+p4c is a reference compiler for the P4 programming language, supporting both P4-14 and P4-16 versions. It is a crucial component in the P4 ecosystem, enabling developers to compile P4 programs into target-specific configurations for efficient network programming and deployment.
+
 [![Main Build](https://github.com/p4lang/p4c/actions/workflows/ci-test-debian.yml/badge.svg)](https://github.com/p4lang/p4c/actions/workflows/ci-test-debian.yml)
 [![Main Build](https://github.com/p4lang/p4c/actions/workflows/ci-test-fedora.yml/badge.svg)](https://github.com/p4lang/p4c/actions/workflows/ci-test-fedora.yml)
 [![Main Build](https://github.com/p4lang/p4c/actions/workflows/ci-test-mac.yml/badge.svg)](https://github.com/p4lang/p4c/actions/workflows/ci-test-mac.yml)
 [![Bazel Build](https://github.com/p4lang/p4c/actions/workflows/ci-bazel.yml/badge.svg)](https://github.com/p4lang/p4c/actions/workflows/ci-bazel.yml)
 [![Validation](https://github.com/p4lang/p4c/actions/workflows/ci-validation-nightly.yml/badge.svg)](https://github.com/p4lang/p4c/actions/workflows/ci-validation-nightly.yml)
 [![Docker Container](https://github.com/p4lang/p4c/actions/workflows/ci-container-image.yml/badge.svg)](https://github.com/p4lang/p4c/actions/workflows/ci-container-image.yml)
 
+
+Table of Contents
+=================
+
+* [p4c](#p4c)
+* [Getting started](#getting-started)
+   * [Installing packaged versions of p4c](#installing-packaged-versions-of-p4c)
+      * [Ubuntu](#ubuntu)
```

Comment:
I would not go that deep, one level of nesting is sufficient.

Reply:
Sure

Reply:
https://github.com/p4lang/p4c/tree/main/testdata do add information about testdata too in TOC @AdarshRawat1.

Reply:
@Abhinavcode13 are you referring to  [Test section in README](https://github.com/AdarshRawat1/p4c/tree/Docs-11-Adding-Table-of-content#tests) ??

Context -  I have covered it in TOC, but will remove it now as we are going to cover only 1 level of nesting.

> TOC is supposed to cover only the contents of README, so we won't be including links to directory in TOC.